packages/core/lib/commands/compile.js

```diff
@@ -98,28 +97,55 @@ const command = {
   },
 
   listVersions: async function (options) {
-    const {CompilerSupplier} = require("@truffle/compile-solidity");
+    const { CompilerSupplier } = require("@truffle/compile-solidity");
+    const { asyncTake } = require("iter-tools");
+
     const supplier = new CompilerSupplier({
-      solcConfig: options.compilers.solc,
+      solcConfig: {
+        ...options.compilers.solc,
+        docker: options.list === "docker"
+      },
       events: options.events
     });
 
     const log = options.logger.log;
     options.list = options.list.length ? options.list : "releases";
 
+    const {
+      latestRelease,
+      releases,
+      prereleases
+    } = await supplier.list();
+    if (options.list === "latestRelease") {
+      log(format(latestRelease, null, " "));
+      return;
+    }
+
+    const allVersions = options.list === "prereleases" ? prereleases : releases;
+    const versions = options.all ? allVersions : asyncTake(10, allVersions);
+
+    if (options.all && options.list === "docker") {
+      log(
+        "Warning: using `--all` with `--list=docker` is very slow and makes " +
+          "many HTTP requests."
+      );
+      log(
+        "You may instead want to browse tags on the web here: " +
+          "https://hub.docker.com/r/ethereum/solc/tags/"
+      );
+    }
+
+    const tags = [];
+    for await (const version of versions) {
+      tags.push(version);
+    }
+
     // Docker tags
-    if (options.list === "docker") {
-      const tags = await supplier.getDockerTags();
+    if (options.list === "docker" && !options.all) {
       tags.push("See more at: hub.docker.com/r/ethereum/solc/tags/");
```

Comment:
Not sure? The reason this is here is because previously there was no way to list `--all` for Docker. I've changed that, so now `truffle compile --list=docker --all` will work, but it'll print a warning to note that the operation is quite slow. (The difference is: solc gives us a single JSON response with all versions, but Docker paginates their results... and there are >500 pages of results)

Reply:
Hm, whatever I guess!